aspnetcore/blazor/tutorials/movie-database-app/part-3.md

```diff
@@ -0,0 +1,585 @@
+
+The at symbol (`@`) with parentheses (`@(...)`), which is called an *explicit Razor expression*, allows the `href` of each link to include the movie entity's `Id` property in the link query string as an *interpolated string* (`$...{...}...`). For a movie identifier (`Id`) of 7, the string value provided to the `href` to edit that movie is `movies/edit?id=7`. When the link is followed, the `id` field is read from the query string by the `Edit` component to load the movie.
+
+```razor
+<QuickGrid Class="table" Items="DB.Movie">
+    <PropertyColumn Property="movie => movie.Title" />
+    <PropertyColumn Property="movie => movie.ReleaseDate" />
+    <PropertyColumn Property="movie => movie.Genre" />
+    <PropertyColumn Property="movie => movie.Price" />
+
+    <TemplateColumn Context="movie">
+        <a href="@($"movies/edit?id={movie.Id}")">Edit</a> |
+        <a href="@($"movies/details?id={movie.Id}")">Details</a> |
+        <a href="@($"movies/delete?id={movie.Id}")">Delete</a>
+    </TemplateColumn>
+</QuickGrid>
+```
+
+For the movie example from the last part of the tutorial series, *The Matrix*&copy;, the <xref:Microsoft.AspNetCore.Components.QuickGrid> component renders the following HTML markup (some elements and attributes aren't present to simplify display). See how the explicit Razor expressions and interpolated strings produced the `href` values for the links to other pages. The movie entity's `Id` is `3`, which is composed in the query strings for the `Edit`, `Details`, and `Delete` pages.
+
+> [!NOTE]
+> The value `3` in the following example is arbitrary and may be different from what you see when you examine the page source of your running app. The number merely matches whatever movie `Id` *The Matrix*&copy; movie has in the database at the time the app is run.
+
+```html
+<table>
+    <thead>
+        <tr>
+            <th>Title</th>
+            <th>ReleaseDate</th>
+            <th>Genre</th>
+            <th>Price</th>
+            <th></th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>The Matrix</td>
+            <td>3/29/1999</td>
+            <td>Sci-fi (Cyberpunk)</td>
+            <td>5.00</td>
+            <td>
+                <a href="movies/edit?id=3">Edit</a> |
+                <a href="movies/details?id=3">Details</a> |
+                <a href="movies/delete?id=3">Delete</a>
+            </td>
+        </tr>
+    </tbody>
+</table>
+```
+
+The column names are taken from the `Movie` model properties, so the release date doesn't have a space between the words. Add a <xref:Microsoft.AspNetCore.Components.QuickGrid.ColumnBase%601.Title> to the <xref:Microsoft.AspNetCore.Components.QuickGrid.PropertyColumn%602> with a value that includes a space between the words:
+
+```razor
+<PropertyColumn Property="movie => movie.ReleaseDate" Title="Release Date" />
+```
+
+Run the app to see that the column now displays two words for the release date.
```

Comment:
I think we also should tell the user to make a similar update in the Create, Edit, Details, and Delete pages to add a space in "Release Date".